core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -99,4 +162,50 @@ public ManagedChannelImpl build() {
    */
   @Internal
   protected abstract ClientTransportFactory buildTransportFactory();
+
+  private static class DirectAddressNameResolver implements NameResolver {
+    final SocketAddress address;
+
+    DirectAddressNameResolver(SocketAddress address) {
+      this.address = address;
+    }
+
+    @Override
+    public Handle resolve(String scheme, @Nullable String nsAuthority, String endpoint) {
+      return new Handle() {
+        @Override
+        public String getServiceAuthority() {
+          if (address instanceof InetSocketAddress) {
```

Comment:
I added `AuthorityOverridingTransportFactory` to `AbstractManagedChannelImplBuilder` to handle authority overriding. Individual builders would not need to implement it.